src/Pact/Analyze/AnalyzeNumerical.hs

```diff
@@ -0,0 +1,185 @@
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE Rank2Types            #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeOperators         #-}
+module Pact.Analyze.AnalyzeNumerical where
+
+import           Control.Lens               (over)
+import           Control.Monad.Except       (MonadError)
+import           Data.SBV                   (EqSymbolic ((.==)), SymWord,
+                                             sDiv, sMod, (.^))
+
+import           Pact.Analyze.Errors
+import           Pact.Analyze.Numerical
+import           Pact.Analyze.Types         hiding (tableName)
+
+class (MonadError AnalyzeFailure m, S :<: TermOf m) => Analyzer m where
```

Comment:
this class is a key to this change

Reply:
very glad to be consolidating this into one class now